### PR TITLE
Promote builds to master and release rather than alpha, beta, and stable

### LINKED ISSUE #1885: Update promote pipeline to match the release process
Currently, the `release/promote` pipeline in Jenkins will call the `promote` target on the `build/release/Makefile`, which operates on all images (Ceph, CockroachDB, Minio) and helm charts.  This means that if we wanted to promote Ceph to the beta channel, then the other storage providers come along for the ride too.  We should be able to only promote specific storage providers instead of all of them.

### PR DESCRIPTION
**Pull request by travisn**
Signed-off-by: travisn <tnielsen@redhat.com>

**Description of your changes:**
The `alpha`, `beta`, and `stable` channels do not match the rook release process. Each storage provider defines their own stability based on the CRDs rather than defining it with the release process. Rook really only has two release streams: master and the official releases. Master is published with each merge, while releases go through a signoff process to ensure quality. Thus, the `alpha`, `beta`, and `stable` channels are removed and replaced with a single `release` channel. See [the detailed notes on this change](https://github.com/rook/rook/issues/1885#issuecomment-460444455).

The `release` channel is only used for promoting the helm chart. The docker images do not receive a channel tag so the versioning can be controlled very carefully for the stateful upgrades.

@bassam @jbw976 PTAL

**Which issue is resolved by this Pull Request:**
Resolves #1885

**Checklist:**
- [x] Documentation has been updated, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

### CODE REVIEW

**Comment by bassam** [comment]
Makes sense on reducing the channels to just `release` and `master`.

> Promoting is now done automatically both for master and release branches. There is no longer a need for a separate promote pipeline.

I'm not sure I follow the reasoning for this. Can you elaborate on why this is needed? Historically we kept a manual promote to ensure that we only promote builds that are fully vetted.

**Comment by travisn** [comment]
>> Promoting is now done automatically both for master and release branches. There is no longer a need for a separate promote pipeline.

> I'm not sure I follow the reasoning for this. Can you elaborate on why this is needed? Historically we kept a manual promote to ensure that we only promote builds that are fully vetted.

I see now that we still need to have the promote step separate for release builds, so I'll revert that. The side effect I overlooked was that the helm chart would be published to the release channel before a new release build would be tagged.

File: build/release/Makefile [hunk 1]
```diff
@@ -160,16 +160,14 @@ build.image.$(1).$(2).$(3):
...
 publish.image.$(1).$(2).$(3): ; @docker push $(1)/$(2)-$(3):$(VERSION)
 publish.all.images: publish.image.$(1).$(2).$(3)
+# tag the master image, but do not tag the release image with a generic channel tag
 promote.image.$(1).$(2).$(3):
 	@docker pull $(1)/$(2)-$(3):$(VERSION)
-	@[ "$(CHANNEL)" = "master" ] || docker tag $(1)/$(2)-$(3):$(VERSION) $(1)/$(2)-$(3):$(VERSION)-$(CHANNEL)
 	@docker tag $(1)/$(2)-$(3):$(VERSION) $(1)/$(2)-$(3):$(CHANNEL)
-	@[ "$(CHANNEL)" = "master" ] || docker push $(1)/$(2)-$(3):$(VERSION)-$(CHANNEL)
-	@docker push $(1)/$(2)-$(3):$(CHANNEL)
+	@[ "$(CHANNEL)" = "release" ] || docker push $(1)/$(2)-$(3):$(CHANNEL)
```
Comment (bassam) [inline build/release/Makefile]: I don't follow why these changes are needed.

File: build/release/Makefile [hunk 2]
```diff
@@ -160,16 +160,14 @@ build.image.$(1).$(2).$(3):
...
 promote.all.images: promote.image.$(1).$(2).$(3)
 clean.image.$(1).$(2).$(3):
 	@[ -z "$$$$(docker images -q $(1)/$(2)-$(3):$(VERSION))" ] || docker rmi $(1)/$(2)-$(3):$(VERSION)
-	@[ -z "$$$$(docker images -q $(1)/$(2)-$(3):$(VERSION)-$(CHANNEL))" ] || docker rmi $(1)/$(2)-$(3):$(VERSION)-$(CHANNEL)
```
Comment (bassam) [inline build/release/Makefile]: why was this removed?

File: build/release/Makefile [hunk 3]
```diff
@@ -178,9 +176,9 @@ $(foreach r,$(REGISTRIES), $(foreach i,$(IMAGES), $(foreach a,$(IMAGE_ARCHS),$(e
 publish.manifest.image.%: publish.all.images $(MANIFEST_TOOL)
 	@$(MANIFEST_TOOL) push from-args --platforms $(IMAGE_PLATFORMS) --template $(DOCKER_REGISTRY)/$*-ARCH:$(VERSION) --target $(DOCKER_REGISTRY)/$*:$(VERSION)
 
+# do not promote images for the release channel
 promote.manifest.image.%: promote.all.images $(MANIFEST_TOOL)
-	@[ "$(CHANNEL)" = "master" ] || $(MANIFEST_TOOL) push from-args --platforms $(IMAGE_PLATFORMS) --template $(DOCKER_REGISTRY)/$*-ARCH:$(VERSION) --target $(DOCKER_REGISTRY)/$*:$(VERSION)-$(CHANNEL)
-	@$(MANIFEST_TOOL) push from-args --platforms $(IMAGE_PLATFORMS) --template $(DOCKER_REGISTRY)/$*-ARCH:$(VERSION) --target $(DOCKER_REGISTRY)/$*:$(CHANNEL)
```
Comment (bassam) [inline build/release/Makefile]: why was this changed?

File: build/release/Makefile (hunk 1 above)
Comment (travisn) [reply build/release/Makefile]: The background for these changes is in the details of [these notes](https://github.com/rook/rook/issues/1885#issuecomment-460444455). To summarize:
- Since there is only a single release channel, there is no need for the channel suffix on the image name. For example, why do we need `v1.0.0-release`?
- Regarding removal of the `release` tag (for example `rook/ceph:release`): To carefully control the versioning of upgrades, it is not recommended to use a tag that changes be used. This is one of the reasons why the `master` tag is regularly a problem for users... because they don't realize when the image changes and it breaks things.

File: build/release/Makefile (hunk 2 above)
Comment (travisn) [reply build/release/Makefile]: Same as above

File: build/release/Makefile (hunk 3 above)
Comment (travisn) [reply build/release/Makefile]: For the reason described above, docker images for the release build are not promoted, we just leave them with the specific release version tag such as `v1.0.0`, which was already done by the publish stage in the pipeline.

**Comment by bassam** [comment]
@travisn and I just chatted about breaking this PR into two parts 1) collapsing alpha, beta, stable channels into just one `release` channel, and 2) removing the ability to pull the latest from each channel. 2) still seems useful to me, but we can discuss it in that PR.

**Comment by travisn** [comment]
@bassam for #2 see #3078, thanks for the discussion
